CMakeLists.txt

```diff
@@ -1,281 +1,294 @@
-cmake_minimum_required(VERSION 2.8)
-project(libevhtp)
+cmake_minimum_required(VERSION 3.1)
 
-set (CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
-include (version)
-include (colors)
-include (options)
-include (CheckFunctionExists)
-include (CheckIncludeFiles)
-include (CheckTypeSize)
-include (CheckCCompilerFlag)
-include (TestBigEndian)
-
-add_definitions ("-DPROJECT_VERSION=${PROJECT_VERSION} -Wall")
+include(HunterGate)
+HunterGate(
+    URL "https://github.com/ruslo/hunter/archive/v0.20.23.tar.gz"
+    SHA1 "e1d36aca5049a11e0dc2460965cecb302120e365"
+)
 
-set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -ggdb3")
-SET(CMAKE_BUILD_TYPE Release CACHE STRING "default to Release")
+project(libevhtp VERSION 1.2.16)
 
-CHECK_FUNCTION_EXISTS(strndup HAVE_STRNDUP)
-CHECK_FUNCTION_EXISTS(strnlen HAVE_STRNLEN)
+include(colors)
+include(options)
+include(CheckFunctionExists)
+include(CheckIncludeFiles)
+include(CheckTypeSize)
+include(CheckCCompilerFlag)
+include(TestBigEndian)
 
-if (EVHTP_THR_SHARED_PIPE)
-    add_definitions(-DEVTHR_SHARED_PIPE)
-endif()
+check_function_exists(strndup HAVE_STRNDUP)
+check_function_exists(strnlen HAVE_STRNLEN)
 
-check_include_files (stdlib.h    HAVE_STDLIB_H)
-check_include_files (string.h    HAVE_STRING_H)
-check_include_files (stdint.h    HAVE_STDINT_H)
-check_include_files (errno.h     HAVE_ERRNO_H)
-check_include_files (strings.h   HAVE_STRINGS_H)
-check_include_files (inttypes.h  HAVE_INTTYPES_H)
-check_include_files (limits.h    HAVE_LIMITS_H)
+check_include_files(stdlib.h    HAVE_STDLIB_H)
```

Comment:
It will be really hard to cherry-pick this change, are you sure you want to do style fixing here? (unless you're not sending this patch upstream?)

Reply:
I have another branch for upstream. Very similar changes.

Reply:
I cheated here and didn't use a standard cherry-pick etc. to apply to this version of libevhtp. I essentially did this: `git checkout origin/cmake-improvements -- $(git diff --name-only origin/develop...origin/cmake-improvements`. That gave me local copies of all the changed CMake files. Then, I committed them into this branch.

Reply:
Let me close and redo this.

Reply:
> I cheated here and didn't use a standard cherry-pick etc. to apply to this version of libevhtp. I essentially did this:

You can loose changes from upstream in this case.